File: jna/src/main/java/dark/metadata.java
package dark;
import com.sun.jna.Pointer;
import com.sun.jna.Structure;

import java.util.Arrays;
import java.util.List;

public class metadata extends Structure implements Structure.ByValue {
    public int classes;
    public Pointer names; //char**

    public metadata() {
        super();
    }

    public metadata(Pointer peer) {
        super(peer);
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("classes","names");
    }
}
/*// option_list.h
typedef struct metadata {
    int classes;
    char **names;
} metadata;
*/

File: jna/src/main/java/dark_kotlin/metadata.kt
package com.sl.microservice.ai.runtime.impl.darknet.dark

import com.sun.jna.Pointer
import com.sun.jna.Structure

class metadata : Structure, Structure.ByValue {
    @JvmField var classes = 0
    @JvmField var names: Pointer? = null //char**

    constructor() : super()
    constructor(peer: Pointer) : super(peer)

    override fun getFieldOrder(): List<String> {
        return listOf("classes", "names")
    }
}
/*// option_list.h
typedef struct metadata {
    int classes;
    char **names;
} metadata;
*/
